### PR TITLE
update storage hasher

### PR DESCRIPTION
**Pull request by ermalkaleci**

### CODE REVIEW

File: modules/synthetic-liquidity-pools/src/lib.rs [hunk 1]
```diff
@@ -37,9 +37,9 @@ pub trait Trait: system::Trait {
 decl_storage! {
 	trait Store for Module<T: Trait> as SyntheticLiquidityPools {
 		pub NextPoolId get(fn next_pool_id): LiquidityPoolId;
-		pub Owners get(fn owners): map hasher(blake2_128_concat) LiquidityPoolId => Option<(T::AccountId, LiquidityPoolId)>;
-		pub LiquidityPoolOptions get(fn liquidity_pool_options): double_map hasher(blake2_128_concat) LiquidityPoolId, hasher(blake2_128_concat) CurrencyId => Option<SyntheticLiquidityPoolOption>;
-		pub Balances get(fn balances): map hasher(blake2_128_concat) LiquidityPoolId => Balance;
+		pub Owners get(fn owners): map hasher(twox_64_concat) LiquidityPoolId => Option<(T::AccountId, LiquidityPoolId)>;
+		pub LiquidityPoolOptions get(fn liquidity_pool_options): double_map hasher(twox_64_concat) LiquidityPoolId, hasher(blake2_128_concat) CurrencyId => Option<SyntheticLiquidityPoolOption>;
```
Comment (xlc) [inline modules/synthetic-liquidity-pools/src/lib.rs:41]:
```suggestion
		pub LiquidityPoolOptions get(fn liquidity_pool_options): double_map hasher(twox_64_concat) LiquidityPoolId, hasher(twox_64_concat) CurrencyId => Option<SyntheticLiquidityPoolOption>;
```

File: modules/synthetic-liquidity-pools/src/lib.rs [hunk 2]
```diff
@@ -37,9 +37,9 @@ pub trait Trait: system::Trait {
...
+		pub Balances get(fn balances): map hasher(twox_64_concat) LiquidityPoolId => Balance;
 		pub MinAdditionalCollateralRatio get(fn min_additional_collateral_ratio) config(): Permill;
 		pub MaxSpread get(fn max_spread): map hasher(blake2_128_concat) CurrencyId => Permill;
```
Comment (xlc) [inline modules/synthetic-liquidity-pools/src/lib.rs:44]:
```suggestion
		pub MaxSpread get(fn max_spread): map hasher(twox_64_concat) CurrencyId => Permill;
```

File: modules/synthetic-liquidity-pools/src/lib.rs (hunk 1 above)
Comment (ermalkaleci) [reply modules/synthetic-liquidity-pools/src/lib.rs:41]: @xlc but CurrencyId is added by the user. would it be safe?

Comment (xlc) [reply modules/synthetic-liquidity-pools/src/lib.rs:41]: yes. CurrencyId is an enum with very limited options. Any invalid CurrencyId will be rejected as invalid transaction during decode phase.

Also it is one byte, so max 256 possibilities, which is safe to hash collision resistance

Comment (ermalkaleci) [reply modules/synthetic-liquidity-pools/src/lib.rs:41]: ok

File: modules/synthetic-tokens/src/lib.rs [hunk 3]
```diff
@@ -43,10 +41,10 @@ impl Default for Position {
...
 decl_storage! {
 	trait Store for Module<T: Trait> as SyntheticTokens {
-		ExtremeRatio get(extreme_ratio): map hasher(blake2_128_concat) CurrencyId => Option<Permill>;
-		LiquidationRatio get(liquidation_ratio): map hasher(blake2_128_concat) CurrencyId => Option<Permill>;
-		CollateralRatio get(collateral_ratio): map hasher(blake2_128_concat) CurrencyId => Option<Permill>;
-		Positions get(positions): double_map hasher(blake2_128_concat) LiquidityPoolId, hasher(blake2_128_concat) CurrencyId => Position;
+		ExtremeRatio get(extreme_ratio): map hasher(twox_64_concat) CurrencyId => Option<Permill>;
+		LiquidationRatio get(liquidation_ratio): map hasher(twox_64_concat) CurrencyId => Option<Permill>;
+		CollateralRatio get(collateral_ratio): map hasher(twox_64_concat) CurrencyId => Option<Permill>;
+		Positions get(positions): double_map hasher(blake2_128_concat) LiquidityPoolId, hasher(twox_64_concat) CurrencyId => Position;
```
Comment (xlc) [inline modules/synthetic-tokens/src/lib.rs:47]:
```suggestion
		Positions get(positions): double_map hasher(twox_64_concat) LiquidityPoolId, hasher(twox_64_concat) CurrencyId => Position;
```

File: modules/margin-protocol/src/lib.rs [hunk 4]
```diff
@@ -75,12 +75,10 @@ pub struct RiskThreshold {
...
-		Positions get(positions): map hasher(blake2_128_concat) PositionId => Option<Position<T>>;
+		Positions get(positions): map hasher(twox_64_concat) PositionId => Option<Position<T>>;
 		PositionsByTrader get(positions_by_trader): double_map hasher(twox_64_concat) T::AccountId, hasher(twox_64_concat) (LiquidityPoolId, PositionId) => Option<()>;
 		PositionsByPool get(positions_by_pool): double_map hasher(twox_64_concat) LiquidityPoolId, hasher(twox_64_concat) (TradingPair, PositionId) => Option<()>;
-		// SwapPeriods get(swap_periods): map hasher(black2_256) TradingPair => Option<SwapPeriod>;
-		Balances get(balances): map hasher(blake2_128_concat) T::AccountId => Fixed128;
-		MinLiquidationPercent get(min_liquidation_percent): map hasher(blake2_128_concat) TradingPair => Fixed128;
+		Balances get(balances): map hasher(twox_64_concat) T::AccountId => Fixed128;
 		MarginCalledTraders get(margin_called_traders): map hasher(blake2_128_concat) T::AccountId => Option<()>;
 		MarginCalledPools get(margin_called_pools): map hasher(blake2_128_concat) LiquidityPoolId => Option<()>;
```
Comment (xlc) [inline modules/margin-protocol/src/lib.rs:83]:
```suggestion
		MarginCalledPools get(margin_called_pools): map hasher(twox_64_concat) LiquidityPoolId => Option<()>;
```

File: modules/margin-protocol/src/lib.rs [hunk 5]
```diff
@@ -75,12 +75,10 @@ pub struct RiskThreshold {
...
 		NextPositionId get(next_position_id): PositionId;
-		Positions get(positions): map hasher(blake2_128_concat) PositionId => Option<Position<T>>;
+		Positions get(positions): map hasher(twox_64_concat) PositionId => Option<Position<T>>;
 		PositionsByTrader get(positions_by_trader): double_map hasher(twox_64_concat) T::AccountId, hasher(twox_64_concat) (LiquidityPoolId, PositionId) => Option<()>;
 		PositionsByPool get(positions_by_pool): double_map hasher(twox_64_concat) LiquidityPoolId, hasher(twox_64_concat) (TradingPair, PositionId) => Option<()>;
-		// SwapPeriods get(swap_periods): map hasher(black2_256) TradingPair => Option<SwapPeriod>;
-		Balances get(balances): map hasher(blake2_128_concat) T::AccountId => Fixed128;
-		MinLiquidationPercent get(min_liquidation_percent): map hasher(blake2_128_concat) TradingPair => Fixed128;
+		Balances get(balances): map hasher(twox_64_concat) T::AccountId => Fixed128;
 		MarginCalledTraders get(margin_called_traders): map hasher(blake2_128_concat) T::AccountId => Option<()>;
```
Comment (xlc) [inline modules/margin-protocol/src/lib.rs:82]:
```suggestion
		MarginCalledTraders get(margin_called_traders): map hasher(twox_64_concat) T::AccountId => Option<()>;
```

Comment (xlc) [reply modules/margin-protocol/src/lib.rs:82]: it is impossible to margin call a non-exist account so this is safe

**Pull request merged**